domains/client/domain-operator/src/domain_block_processor.rs

```diff
@@ -659,9 +663,37 @@ where
         &self,
         receipts: Vec<ExecutionReceiptFor<Block, ParentChainBlock>>,
         fraud_proofs: Vec<FraudProof<NumberFor<ParentChainBlock>, ParentChainBlock::Hash>>,
+        invalid_bundles: Vec<InvalidBundle<ParentChainBlock::Hash>>,
     ) -> Result<(), sp_blockchain::Error> {
         let mut bad_receipts_to_write = vec![];
 
+        // write invalid bundles with invalid ER first. If the state transition itself was invalid on these
+        // ERs, then they will be replaced by the next loop.
+        for invalid_bundle in invalid_bundles {
+            if let InvalidBundleType::InvalidReceipt(InvalidReceipt::InvalidTotalRewards {
+                consensus_block_hash,
+                bad_receipt_hash,
+            }) = invalid_bundle.invalid_bundle_type
```

Comment:
We don't need to pass the `invalid_bundles` along from `block-preprocessor` here, because the `receipts` argument already contains the invalid reward ER, thus they can fit in the same flow as the invalid state transition ER as below.